cmd/ci-secret-bootstrap/main.go

```diff
@@ -0,0 +1,297 @@
+				if err != nil {
+					return nil, err
+				}
+				field, err := itemGetter.GetField(bwContext.Field)
+				if err != nil {
+					return nil, err
+				}
+				value = []byte(field)
+			} else {
+				itemGetter, err := bwClient.GetItem(bwContext.BWItem)
+				if err != nil {
+					return nil, err
+				}
+				value, err = itemGetter.GetAttachment(bwContext.Attachment)
+				if err != nil {
+					return nil, err
+				}
+			}
+			if err != nil {
+				return nil, err
+			}
+			data[key] = value
+		}
+		for _, secretContext := range secretConfig.To {
+			secret := &coreapi.Secret{
+				Data: data,
+				ObjectMeta: meta.ObjectMeta{
+					Name:      secretContext.Name,
+					Namespace: secretContext.Namespace,
+					Labels:    map[string]string{"ci.openshift.org/auto-managed": "true"},
+				},
+			}
+			secretsMap[secretContext.Cluster] = append(secretsMap[secretContext.Cluster], secret)
+		}
+	}
+	return secretsMap, nil
+}
+
+func applySecrets(secretsGetters map[string]coreclientset.SecretsGetter, secretsMap map[string][]*coreapi.Secret) error {
+	for cluster, secrets := range secretsMap {
+		for _, secret := range secrets {
+			secretsGetter := secretsGetters[cluster]
+			_, err := secretsGetter.Secrets(secret.Namespace).Create(secret)
+			if err != nil {
+				if !kerrors.IsAlreadyExists(err) {
+					return err
+				}
+				if _, err := secretsGetter.Secrets(secret.Namespace).Update(secret); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func dryRunDump(secretsGetters map[string]coreclientset.SecretsGetter, w io.Writer) error {
+	var clusters []string
+	for cluster := range secretsGetters {
+		clusters = append(clusters, cluster)
+	}
+	sort.Strings(clusters)
+	for _, cluster := range clusters {
+		if _, err := fmt.Fprintln(w, fmt.Sprintf("###%s###", cluster)); err != nil {
+			return err
+		}
+		secrets, err := secretsGetters[cluster].Secrets("").List(meta.ListOptions{})
+		if err != nil {
+			return err
+		}
+		s := json.NewSerializerWithOptions(json.DefaultMetaFactory, scheme.Scheme,
+			scheme.Scheme, json.SerializerOptions{Yaml: true, Pretty: true, Strict: true})
+		err = s.Encode(secrets, w)
+	}
+	return nil
+}
+
+func main() {
+	o := parseOptions()
+	if err := o.validateOptions(); err != nil {
+		logrus.WithError(err).Fatal("Invalid arguments.")
+	}
+	if err := o.completeOptions(); err != nil {
+		logrus.WithError(err).Fatal("Failed to complete options.")
+	}
+	bwClient, err := bitwarden.NewBitwardenClient(o.bwUser, o.bwPassword)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to get Bitwarden client.")
+	}
+	defer func() {
+		if err := bwClient.Logout(); err != nil {
+			logrus.WithError(err).Fatal("Failed to logout Bitwarden.")
+		}
+	}()
+
+	secretsMap, err := getSecrets(o.config, bwClient)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to get secrets.")
+	}
+
+	if o.dryRun {
+		for cluster := range o.secretsGetters {
+			o.secretsGetters[cluster] = fake.NewSimpleClientset().CoreV1()
```

Comment:
Oh my. No. Let's not use a test fake here. Just print the things.

Reply:
Removed.